include/zephyr/bluetooth/audio/has.h

```diff
@@ -95,6 +102,57 @@ struct bt_has_client_cb {
 	 * @param index Active preset index.
 	 */
 	void (*preset_switch)(struct bt_has *has, uint8_t index);
+
+	/**
+	 * @brief Callback function for presets read operation.
+	 *
+	 * The callback called when the preset record update is notified by the remote server.
+	 *
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param err 0 on success, ATT error or negative errno otherwise.
+	 * @param record Preset record or NULL on errors.
+	 * @param is_last True if Read Presets operation can be considered concluded.
+	 */
+	void (*preset_read_rsp)(struct bt_has *has, int err,
+				const struct bt_has_preset_record *record, bool is_last);
```

Comment:
Would it make sense that instead of supplying the `is_last` to the application, that we somehow collect and aggregate all the responses in an array and provide a single callback? 
I don't suppose the client can in any way stop the "read" (which is, IIRC, a multiple of notifications, right?)?

Reply:
I guess that would require having additional buffer for this that might increase the memory footprint unnecesarily. I don't feel this is needed

Reply:
Agreed